libraries/Microsoft.Bot.Connector/Teams/TeamsOperations.cs

```diff
@@ -899,6 +929,148 @@ private async Task<HttpOperationResponse<string>> SendMessageToListOfUsersWithRe
             return result;
         }
 
+        private async Task<HttpOperationResponse<string>> GetOperationStateWithRetryAsync(string operationId, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            // Tracing
+            var shouldTrace = ServiceClientTracing.IsEnabled;
+            string invocationId = null;
+            if (shouldTrace)
+            {
+                invocationId = ServiceClientTracing.NextInvocationId.ToString(CultureInfo.InvariantCulture);
+                var tracingParameters = new Dictionary<string, object>();
+                tracingParameters.Add("operationId", operationId);
+                tracingParameters.Add("cancellationToken", cancellationToken);
+                ServiceClientTracing.Enter(invocationId, this, "GetOperationState", tracingParameters);
+            }
+
+            // Construct URL
+            var baseUrl = Client.BaseUri.AbsoluteUri;
+            var url = new Uri(new Uri(baseUrl + (baseUrl.EndsWith("/", StringComparison.InvariantCulture) ? string.Empty : "/")), "v3/batch/conversation/" + operationId).ToString();
+            using var httpRequest = new HttpRequestMessage();
```

Comment:
Check how other methods in this class handle URI parameters. I think they create the URL with {operationId} and then replace it with the value. It's for clarity.